apps/admin/src/actions/user.ts

```diff
@@ -96,11 +96,19 @@ export async function updateUser(data: z.infer<typeof baseInformationSchema>, id
 export async function updateUserState(data: z.infer<typeof userStateSchema>, id: string) {
     await checkIfAdmin();
 
+    if (data.experimentStatus === "installation" && !data.installationDate) {
+        throw new Error("Installation date is required for installation status");
+    }
+
+    if (data.experimentStatus === "deinstallation" && !data.deinstallationDate) {
+        throw new Error("Deinstallation date is required for deinstallation status");
+    }
+
     try {
         await updateUserDb(
             {
+                isActive: data.isActive,
```

Comment:
Müsstest du hier nicht auch irgendwo `data.experimentStatus` reingeben?
Ansonsten stellt sich mir die Frage, warum das in `userStateSchema` drin ist

Reply:
Ist alles WIP, da ja doch einiges noch eingebaut werden muss 👯